Sources/Linkability/Analyze.swift

```diff
@@ -0,0 +1,263 @@
+        gTLDLinked: gTLDLinked,
+        gTLDBrandLinked: gTLDBrandLinked,
+        gTLDNonBrandLinked: gTLDNonBrandLinked
+    )
+}
+
+public enum ZoneType {
+    case ccTLD
+    case gTLDAll
+    case gTLDBrand
+    case gTLDNonBrand
+}
+
+public func getLinkedZones(type: ZoneType) {
+    let zones = readZones(from: "Data-Zones/zones-full.txt")
+    if zones.isEmpty {
+        print("No zones found in Data-Zones/zones-full.txt")
+        return
+    }
+
+    let brandZones = Set(readZones(from: "Data-Zones/zones-brand.txt"))
+    var linkedZones: [String] = []
+
+    for zone in zones {
+        let isLinked = isZoneLinked(zone)
+        if !isLinked { continue }
+
+        let isCCTLD = zone.count == 2 && zone.allSatisfy({ $0.isASCII })
+        let isBrand = brandZones.contains(zone)
+
+        switch type {
+        case .ccTLD:
+            if isCCTLD {
+                linkedZones.append(zone)
+            }
+        case .gTLDAll:
+            if !isCCTLD {
+                linkedZones.append(zone)
+            }
+        case .gTLDBrand:
+            if !isCCTLD && isBrand {
+                linkedZones.append(zone)
+            }
+        case .gTLDNonBrand:
+            if !isCCTLD && !isBrand {
+                linkedZones.append(zone)
+            }
+        }
+    }
+
+    let typeDescription =
+        switch type {
+        case .ccTLD: "ccTLD"
+        case .gTLDAll: "gTLD"
+        case .gTLDBrand: "brand gTLD"
+        case .gTLDNonBrand: "non-brand gTLD"
+        }
+
+    let sortedZones = linkedZones.sorted()
+    print("\nLinked \(typeDescription) zones (\(sortedZones.count)):")
+
+    if sortedZones.isEmpty {
+        print("No linked \(typeDescription) zones found")
+    } else {
+        print(sortedZones.joined(separator: " "))
+    }
+}
+
+public func showMissingActiveBrands() {
+    print("Checking for delegated brand zones not in root zone file...")
+
+    let zones = readZones(from: "Data-Zones/zones-full.txt")
+    if zones.isEmpty {
+        print("❌ Error: No zones found in Data-Zones/zones-full.txt")
+        return
+    }
+
+    // Read brand zones
+    let brandZones = Set(readZones(from: "Data-Zones/zones-brand.txt"))
+    let rootZones = Set(zones)
+
+    // Find brand zones not in root zone
+    let missingBrands = brandZones.subtracting(rootZones)
+
+    print("")
+    print("Delegated brand zones missing from root zone file:")
+    print("Total delegated brand zones: \(brandZones.count)")
+    print("Delegated brands in root zone: \(brandZones.intersection(rootZones).count)")
+    print("Delegated brands missing from root zone: \(missingBrands.count)")
+    print("")
+
+    if !missingBrands.isEmpty {
+        print("Missing delegated brand zones:")
+        for brand in missingBrands.sorted() {
+            print("  \(brand)")
+        }
+    } else {
+        print("✅ All delegated brand zones are present in the root zone file!")
+    }
+}
+
+public func testCCTLDBrands() {
```

Comment:
Should this be in here or is it actually a unit test?

Reply:
Good catch, I think what I'll do is check the ZoneDB project to see if it has a test for this. (and if not, I'll add one so that we can derive this from the upstream ZoneDB project)